_rules/heading-descriptive-b49b2e.md

```diff
@@ -29,13 +29,16 @@ This rule applies to any element with the [semantic role][] of heading that is e
 
 ## Expectation
 
-Each target element describes the topic or purpose of its [section of the content][].
+Each target element describes the topic or purpose of the [programmatic section of content][] it is [associated][] with.
 
 **Note:** Headings do not need to be lengthy. A word, or even a single character, may suffice.
 
 ## Assumptions
 
-This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content), and sufficiently understood.
+- This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content), and sufficiently understood.
+- This rule assumes that the [semantic roles][semantic role] (including the level of headings) are used according to their semantics.
+- This rule assumes that [nested browsing contexts][nested browsing context] included in an implicit section of content do not contain any element with a role of `heading` and an `aria-level` lower or equal to the one of the `heading` defining this implicit section of content.
+- This rule assumes that the `hgroup` element is not used.
```

Comment:
We should probably just deal with this in the definition, rather than assume it isn't used.

Reply:
The problem is that is has no default role. Handling it will make the definition mixes roles and elements. It is doable, but it gets much more dirty (the "implicit section" delimiter will be `hgroup` or `role="heading"` not within an `hgroup`).

The fact that `hgroup` was deprecated in HTML 5.2 (but kept in WHATWG living standard… 🙄) also played its part in me choosing to just ignore it…

According to https://www.advancedwebranking.com/html/#content-sectioning `hgroup` represents 0.1% of the **Content sectioning** elements used (September 2019). I say we're safe enough in ignoring it.